### PR TITLE
refactor: extract RecipeDetails to a component

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Extract the Application/Recipe details code to a standalone component 

### Screenshot / video of UI

No visual changes

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/251

### How to test this PR?

- Tests in `Recipe.spec.ts` ensure no regression

### CODE REVIEW

File: packages/frontend/src/lib/RecipeDetails.svelte [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { getDisplayName } from '/@/utils/versionControlUtils';
+import type { Recipe } from '@shared/src/models/IRecipe';
+import { recipes } from '/@/stores/recipe';
+import Button from '/@/lib/button/Button.svelte';
+import TasksProgress from '/@/lib/progress/TasksProgress.svelte';
+import Fa from 'svelte-fa';
+import { studioClient } from '/@/utils/client';
+import { catalog } from '/@/stores/catalog';
+
+export let recipeId: string;
```
Comment (jeffmaury) [inline packages/frontend/src/lib/RecipeDetails.svelte:13]: Not required as id is already present in the Recipe type

**Review by jeffmaury — CHANGES_REQUESTED** [review]

File: packages/frontend/src/lib/RecipeDetails.svelte (hunk 1 above)
Comment (axel7083) [reply packages/frontend/src/lib/RecipeDetails.svelte:13]: In the recipe interface, the property id is optional, therefore we either need to make a huge {#try}{/if} or improve the Recipe Interface to make he id  a mandatory property

Comment (axel7083) [reply packages/frontend/src/lib/RecipeDetails.svelte:12]: I fixed it by removing the recipe property @jeffmaury

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**